### PR TITLE
Minor updates and bug fixes for RDS source

### PR DESCRIPTION
**Pull request by oeyh**
### Description
- Sets external origination time for stream events to enable end to end latency metric
- Adds a config to allow disabling primary stream node from doing S3 scan work
- Fixes a bug with parsing ExportObjectKey
- Fixes an NPE when getting TlsConfig
 
### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/converter/StreamRecordConverter.java [hunk 1]
```diff
@@ -63,6 +64,11 @@ public Event convert(final Map<String, Object> rowData,
 
         EventMetadata eventMetadata = event.getMetadata();
 
+        // Only set external origination time for stream events, not export
+        final Instant externalOriginationTime = Instant.ofEpochMilli(eventCreateTimeEpochMillis);
```
Comment (dinujoh) [inline data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/converter/StreamRecordConverter.java:68]: is eventCreateTimeEpochMillis read from the bin logs ?

File: data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/RdsService.java [hunk 2]
```diff
@@ -146,7 +146,7 @@ private SchemaManager getSchemaManager(final RdsSourceConfig sourceConfig, final
                 dbMetadata.getPort(),
                 sourceConfig.getAuthenticationConfig().getUsername(),
                 sourceConfig.getAuthenticationConfig().getPassword(),
-                !sourceConfig.getTlsConfig().isInsecure());
+                !sourceConfig.isTlsEnabled());
```
Comment (dinujoh) [inline data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/RdsService.java:149]: Should this be `sourceConfig.isTlsEnabled()` without `!`

File: data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/converter/StreamRecordConverter.java (hunk 1 above)
Comment (oeyh) [reply data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/converter/StreamRecordConverter.java:68]: Yes, it's from binlog event timestamp: https://github.com/opensearch-project/data-prepper/blob/d6465efc63f98380605b90ddeab90343bb469dbf/data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/BinlogEventListener.java#L224

File: data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/RdsService.java (hunk 2 above)
Comment (oeyh) [reply data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/RdsService.java:149]: Good catch. Will fix.

File: data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/model/ExportObjectKey.java [hunk 3]
```diff
@@ -36,7 +36,9 @@ public static ExportObjectKey fromString(final String objectKeyString) {
         final String prefix = parts[0];
         final String exportTaskId = parts[1];
         final String databaseName = parts[2];
-        final String tableName = parts[3];
+        // fullTableName is in the format of "databaseName.tableName"
+        final String fullTableName = parts[3];
+        final String tableName = fullTableName.split("\\.")[1];
```
Comment (dinujoh) [inline data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/model/ExportObjectKey.java:41]: Should we enforce validation here ? Can there be user error that may lead to NPE ?

**Review by dinujoh — APPROVED** [review]

**Review by chenqi0805 — APPROVED** [review]

**Pull request merged**